examples/webgl_raymarching_reflect.html

```diff
@@ -0,0 +1,351 @@
+				));
+
+			}
+
+			float getShadow( vec3 ro, vec3 rd ) {
+
+				float h = 0.0;
+				float c = 0.0;
+				float r = 1.0;
+				float shadowCoef = 0.5;
+
+				for ( float t = 0.0; t < 50.0; t++ ) {
+
+					h = sceneDist( ro + rd * c );
+
+					if ( h < EPS ) return shadowCoef;
+
+					r = min( r, h * 16.0 / c );
+					c += h;
+
+				}
+
+				return 1.0 - shadowCoef + r * shadowCoef;
+
+			}
+
+			vec3 getRayColor( vec3 origin, vec3 ray, out vec3 p, out vec3 normal, out bool hit ) {
+
+				// marching loop
+				float dist;
+				float depth = 0.0;
+				p = origin;
+
+				for ( int i = 0; i < 64; i++ ){
+
+					dist = sceneDist( p );
+					depth += dist;
+					p = origin + depth * ray;
+
+					if ( abs(dist) < EPS ) break;
+
+				}
+
+				// hit check and calc color
+				vec3 color;
+
+				if ( abs(dist) < EPS ) {
+
+					normal = getNormal(p);
+					float diffuse = clamp( dot( lightDir, normal ), 0.1, 1.0 );
+					float specular = pow( clamp( dot( reflect( lightDir, normal ), ray ), 0.0, 1.0 ), 10.0 );
+					float shadow = getShadow( p + normal * OFFSET, lightDir );
+					color = ( sceneColor( p ).rgb * diffuse + vec3( 0.8 ) * specular ) * max( 0.5, shadow );
+
+					hit = true;
+
+				} else {
+
+					color = vec3( 0.0 );
+
+				}
+
+				return color - pow( clamp( 0.05 * depth, 0.0, 0.6 ), 2.0 );
+
+			}
+
+			void main(void) {
+
+				// fragment position
+				vec2 p = ( gl_FragCoord.xy * 2.0 - resolution ) / min( resolution.x, resolution.y );
+
+				// camera and ray
+				vec3 cPos  = vec3( mouse.x - 0.5, mouse.y * 4.0 - 0.2, time );
+				vec3 cDir  = normalize( vec3( 0.0, -0.3, 1.0 ) );
+				vec3 cUp   = cross( cDir, vec3( 1.0, 0.0 ,0.0 ) );
+				vec3 cSide = cross( cDir, cUp );
+				float targetDepth = 1.3;
+				vec3 ray = normalize( cSide * p.x + cUp * p.y + cDir * targetDepth );
+
+				vec3 color = vec3( 0.0 );
+				vec3 q, normal;
+				bool hit;
+				float alpha = 1.0;
+
+				for ( int i = 0; i < 3; i++ ) {
+
+					color += alpha * getRayColor( cPos, ray, q, normal, hit );
+					alpha *= 0.3;
+					ray = normalize( reflect( ray, normal ) );
+					cPos = q + normal * OFFSET;
+
+					if ( !hit ) break;
+
+				}
+
+				gl_FragColor = vec4(color, 1.0);
+
+			}
+
+		</script>
+
+		<script id="vertex_shader" type="x-shader/x-vertex">
+
+			void main(void) {
+
+				gl_Position = vec4(position, 1.0);
+
+			}
+
+		</script>
+
+		<script src="../build/three.min.js"></script>
+
+		<script src="js/libs/stats.min.js"></script>
+		<script src="js/libs/dat.gui.min.js"></script>
+
+		<script>
+
+			var scene, camera, renderer;
+			var geometry, material, mesh;
+			var mouse = new THREE.Vector2( 0.5, 0.5 );
+			var canvas;
+			var stats;
+
+			var config = {
+				saveImage: function() {
+
+					renderer.render( scene, camera );
+					window.open( canvas.toDataURL() );
+
+				},
+				recursion: 3,
+				resolution: '512',
+			};
+
+			init();
+			render();
+
+			function init() {
+
+				scene = new THREE.Scene();
+				camera = new THREE.Camera();
```

Comment:
Hmmm. I just realized you are instantiating a `THREE.Camera()` and using a simplified vertex shader. This was also done in http://threejs.org/examples/webgl_shader.html. Personally, I would use an orthographic camera to render the quad, but your approach does appear to work...

@mrdoob -- Do you have an opinion on this?